pool/src/transaction_pool.rs

```diff
@@ -198,7 +202,23 @@ where
 		// Check the tx lock_time is valid based on current chain state.
 		self.blockchain.verify_tx_lock_height(tx)?;
 
-		self.blockchain.replay_attack_check(tx)?;
+		{
+			let mut replay_cache = self.replay_verifier_cache.write();
+			let mut tx_need_to_verify = false;
+			let kernel_list = tx.body.kernels.clone();
+			for kernel in kernel_list.clone() {
```

Comment:
The   replay_cache  has kernels, not transactions.
I believe even compact slates allows to keep the kernel and change outputs, right?
So attacker can feed correct transaction to satisfy the cache (this tx can fail later because of some checking that will be apply later). Then attacker will post another tx with reply output and the same kernel.  It will not be verified because of cache
In cache you can have hash from all transaction, it should be safe